### PR TITLE
Fix the unit test errors / enable accuracy tests

### PR DESCRIPTION
**Pull request by nvchenghaoz**
@coderabbitai summary

Several updates

1. Enable the accuracy testing for nemotron-h, added tests for MMLU and gsm8k

2. Fix two unit tests.

### CODE REVIEW

File: tensorrt_llm/_torch/auto_deploy/custom_ops/cuda_backend_causal_conv.py [hunk 1]
```diff
@@ -186,6 +186,12 @@ def _cuda_cached_causal_conv1d(
         # Use true start offsets for decode tokens (tail after prefills)
         decode_idx = seq_start[num_prefill:].to(torch.long)
         x_decode = inp_flat.index_select(0, decode_idx)  # [num_decode, C_in]
+        slot_idx_decode = slot_idx[num_prefill:].to(torch.long)
+        # Only zero-initialize decode cache rows in the mixed/context path.
+        # For generate-only (s == 1), caches must carry prior state.
+        if num_prefill > 0 and slot_idx_decode.numel() > 0:
+            zero_rows = torch.zeros_like(conv_state_cache.index_select(0, slot_idx_decode))
+            conv_state_cache.index_copy_(0, slot_idx_decode, zero_rows)
```
Comment (lucaslie) [inline tensorrt_llm/_torch/auto_deploy/custom_ops/cuda_backend_causal_conv.py:194]: why would we reset cache state for decode?

File: tensorrt_llm/_torch/auto_deploy/custom_ops/triton_backend_mamba.py [hunk 2]
```diff
@@ -124,7 +124,12 @@ def _triton_cached_ssm_transform(
 
     # Decode: batch single-token updates via selective_state_update
     if num_decode > 0:
-        decode_idx = seq_start[num_prefill:].to(torch.long)
+        # In generate-only (s == 1), each batch element has one token and seq_start entries
+        # are typically zeros. Use arange over the flattened batch to index tokens correctly.
+        if s == 1:
+            decode_idx = torch.arange(bs, device=device, dtype=torch.long)
+        else:
+            decode_idx = seq_start[num_prefill:].to(torch.long)
```
[the remark this answers is not in the document]
Comment (nvchenghaoz) [reply tensorrt_llm/_torch/auto_deploy/custom_ops/triton_backend_mamba.py:132]: Probably will keep it as a fallback solution.. 

This is to fix an error in the unit test.
